code/modules/shuttle/shuttle_manipulator.dm

```diff
@@ -160,6 +167,10 @@
 					break
 
 		if("preview")
+			if(shuttle_and_preview_cooldown + PREVIEW_OR_SHUTTLE_SPAWN_COOLDOWN > world.time) 
```

Comment:
I would add some brackets on this to make the logic absolutely sure

Reply:
Cooldown being added is now handled on the application of a new CD (think this is what you were suggesting